### DISCUSSION TITLE
React router v7

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#6588] [General]
Hi there, react router v7 is officially released and is in favor of vite plugin. I am wondering how modern js will be affected by it since Vite plugin offers a lot of features.

thanks :)

**Comment by cloud0816** [comment]
Hi. AsuraKev. I hpoe this answer help you.

The release of React Router v7 with support for Vite plugins marks a shift toward more streamlined and efficient development workflows for modern JavaScript applications. Known for its fast build times and hot module replacement, Vite improves the developer experience by enabling faster iteration and testing. By integrating with Vite, React Router v7 can take advantage of these performance benefits, resulting in faster development and reduced build times for React applications. In addition, the Vite plugin ecosystem offers a wide range of features and tools, making it easier for developers to incorporate advanced features and optimizations into their projects. Overall, this integration highlights a growing trend in the JavaScript ecosystem toward tools that prioritize speed, modularity, and an improved developer experience.

Thank you.

**Comment by zllkjc** [comment]
Let me try to answer this question. React Router v7 provides two access ways, one is to use the library (the original React Router), and the other is to use the framework (Remix). The point you mentioned should refer to using the framework.

We believe that React Router is a very good and popular routing library, so we did not develop our own routing in Modern.js. Modern.js relies on React Router as a library, and the build of Modern.js depends on Rsbuild/Rspack. Except for routing, most of the framework's functions are unrelated to Remix. If there are capabilities that Modern.js currently does not have, we will also implement them based on the logic of ourself.

Of course, because we rely on React Router for routing, there are also some functions that we need to wait for React Router's implementation, such as SPA dimension processing of RSC. We are also continuously paying attention to React Router's implementation in this area.

**Comment by yimingjfe** [comment]
Modern.js supports react-router v7 now, and you can use it by the latest version (`2.67.6` and above) and refer the [doc](https://modernjs.dev/guides/basic-features/routes.html#upgrading-to-react-router-v7).

In the next major version, React Router v7 will be the default option.

**Comment by omarshibli** [comment]
Hey, when the next major version?
I added `@modern-js/plugin-router-v7` but still some hooks from react router v7 are missing, since `@modern-js/runtime/router` still points to react router v6.
